collector/mocks_test.go

```diff
@@ -0,0 +1,1020 @@
+package collector_test
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/norman/clientbase"
+	"github.com/rancher/norman/types"
+	rancherCluster "github.com/rancher/rancher/pkg/client/generated/cluster/v3"
+	rancher "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	rancherProject "github.com/rancher/rancher/pkg/client/generated/project/v3"
+)
+
+func NewClientForTestingNamespace(data string) (*rancherCluster.Client, error) {
+	client := &rancherCluster.Client{}
+	client.Namespace = NewNamespaceOperationsMock(data)
+	return client, nil
+}
+
+func NewClientForTestingWorkload(data string) (*rancherProject.Client, error) {
+	client := &rancherProject.Client{}
+	client.Workload = NewWorkloadOperationsMock(data)
+	return client, nil
+}
+
+func NewTestClientOpts() *clientbase.ClientOpts {
+	ret := &clientbase.ClientOpts{}
+	ret.URL = "test"
+	ret.AccessKey = "test"
+	ret.SecretKey = "test"
+	ret.TokenKey = "test"
+	return ret
+}
+
+type OperationsMockInterface interface {
+	SetData(data string) error
+}
```

Comment:
Apparently this interface can be removed, as it does not seem to be used anywhere.

Reply:
removed